### ISSUE TITLE
strange python indenting with sw=2

### ISSUE THREAD

**Issue by strux-net** [vim/vim#3341]
I start vim (8.1.290) with `vim -u NONE -N -c "filetype plugin on|filetype indent on"`
Then I `:e file.py` with this content
```
def foo():
    a=1
    b=2
```
There are 4 spaces at the start of lines 2 and 3
now I do `:set sw=2 `and `gg=G `to reformat the buffer and I get
```
def foo():
  a=1
    b=2
```
but what I expect is:
```
def foo():
  a=1
  b=2
```

**Comment by chrisbra** [comment]
Yes, currently it keeps the current indent, because of the `return -1` here:
https://github.com/vim/vim/blob/b476cb7d8d1a8c02409f110dea8b166aa9334e18/runtime/indent/python.vim#L191

Perhaps it would be better to always `return plindent` (e.g. the indent of the previous line, if no other condition is met).

Note, current indent maintainer is Bram, so it is his opinion how to handle it.
